### PR TITLE
Daccess 901 pr template

### PR DESCRIPTION
**Pull request by JeremyDuncan**
`Example of how it would look once utilized:`

## 📝 Description

This is a template example that can be fine tuned to our needs. Once it has been merged to the default branch (dev, and main) it will automatically populate this template.

## 🔗 Related Issues

- [DACCESS-901](<https://culibrary.atlassian.net/browse/DACCESS-901>) 

## 🔖 Type of Change

- [X] 🚀 `feature` — New feature or enhancement
- [ ] 🐛 `bug` — Bug fix
- [ ] 🧰 `maintenance` — Maintenance
- [ ] 🦮 `accessibility` — Accessibility updates
- [ ] 📚 `documentation` — Documentation changes

## 🧑‍💻 How Reviewers Can Test This

- Nothing can be tested until the template is committed to the default branch
- Once this PR has been merged into dev, create a PR and see if the Template populates.
- Should see a automated PR template populate with every PR once this has been commited.

## 📸 Screenshots

- N/A

## 🚀 Deployment Notes

- N/A

## ✅ Checklist
- [X] Tests added/updated (if needed)
- [X] Documentation updated (if needed)
- [X] No secrets, API keys, or credentials committed

[DACCESS-901]: https://culibrary.atlassian.net/browse/DACCESS-901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DACCESS-901]: https://culibrary.atlassian.net/browse/DACCESS-901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JeremyDuncan** [comment]
PR Template Documentation:
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

**Review by mhk33 — APPROVED** [review]
this is cool! i'm trying to think what type of change i would select for the accessibility work that i do. maybe enhancement or maintenance? i assume you can select more than one type of change?

**Comment by JeremyDuncan** [comment]
> this is cool! i'm trying to think what type of change i would select for the accessibility work that i do. maybe enhancement or maintenance? i assume you can select more than one type of change?

We could always add an accessibility check box that covers that too?

**Comment by mhk33** [comment]
> > this is cool! i'm trying to think what type of change i would select for the accessibility work that i do. maybe enhancement or maintenance? i assume you can select more than one type of change?
> 
> We could always add an accessibility check box that covers that too?

that would be good!

**Comment by FrancesWebb** [comment]
What kind of breaking change are we imagining would earn the "breaking" change type?

**Comment by JeremyDuncan** [comment]
> What kind of breaking change are we imagining would earn the "breaking" change type?

It could be anything that may affect another service, such as API endpoint changes, or Rails/GEM changes that may affect another service or feature.

It may be something we really don't need and could be removed. I think "Maintenance" is one field that could be removed as well since I think it would be implied any change would be considered maintenance.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+
+
+<!-- ############################## ⚠️ REQUIRED ################################### -->
+## 🔖 Type of Change
+<!-- Add "X" to one or more boxes that apply. (At least one is required) -->
+
+- [ ] 🚀 `feature` — New feature or enhancement
+- [ ] 🐛 `bug` — Bug fix
+- [ ] 🧰 `maintenance` — Maintenance
```
Comment (chrisrlc) [inline .github/pull_request_template.md:20]: I like maintenance but agree with removing breaking. I think we could also fold security into maintenance, but I don't mind keeping them both if others prefer it.

Refactor and cleanup seem also pretty similar - I might fold them all into maintenance as well, or at least just keep 1 of them. Not sure how valuable it is to separate these out.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+
+<!-- ############################## ⚠️ REQUIRED ################################### -->
+## 🏷️ Version Bump
+<!-- Check ONE -->
+
+- [ ] `major` — Breaking changes
+- [ ] `minor` — New features, backward compatible
+- [ ] `patch` — Bug fixes, backward compatible
```
Comment (chrisrlc) [inline .github/pull_request_template.md]: Having trouble understanding how these categories apply to this rails app, rather than an external dependency like a gem or api. What might we be breaking?

**Comment by FrancesWebb** [comment]
If we want to use a matching template for the indexing work, should we have any additional PR types to support that? We will have various changes that will affect what is and isn't in Solr. Also, hypothetically, what would a PR type be for Shinwoo's work on LC authority activity streams? These are important infrastructure improvements, but they will have absolutely no effect on Blacklight until later steps.

File: .github/pull_request_template.md (hunk 2 above)
Comment (JeremyDuncan) [reply .github/pull_request_template.md]: Yeah I see what you mean. I can remove this portion from the template.

**Comment by JeremyDuncan** [comment]
> If we want to use a matching template for the indexing work, should we have any additional PR types to support that? We will have various changes that will affect what is and isn't in Solr. Also, hypothetically, what would a PR type be for Shinwoo's work on LC authority activity streams? These are important infrastructure improvements, but they will have absolutely no effect on Blacklight until later steps.

Thats a good point and I'm sure there may be a few good thoughts on this. Would this not fit under the Maintenance section?

File: .github/pull_request_template.md (hunk 2 above)
Comment (JeremyDuncan) [reply .github/pull_request_template.md]: Removed 👍

File: .github/pull_request_template.md (hunk 1 above)
Comment (JeremyDuncan) [reply .github/pull_request_template.md:20]: I removed Security, Cleanup, and Refactor.  It does make sense to keep the tags a bit slimmed down so the ChangLog categories are more streamlined 👍

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+- [ ] 🚀 `feature` — New feature or enhancement
+- [ ] 🐛 `bug` — Bug fix
+- [ ] 🧰 `maintenance` — Maintenance
+- [ ] 🦮 `accessibility` — Accessibility updates
+- [ ] 📚 `documentation` — Documentation changes
+
+
+
+<!-- ############################## ⚠️ REQUIRED ################################### -->
+## 🧪 How This Was Tested
```
Comment (chrisrlc) [inline .github/pull_request_template.md]: I'm not sure we need both this and the How Reviewers Can Test This section below. Maybe we could just describe edge cases for testing in the section below for reviewers? We already have a checklist item for adding tests and feels tedious to list them out in this PR description when the PR already includes code changes.

Comment (JeremyDuncan) [reply .github/pull_request_template.md]: Thats a good point! ☝️

Comment (JeremyDuncan) [reply .github/pull_request_template.md]: ok removed and generalized the review section to make it less tedious and more streamlined

**Review by chrisrlc — APPROVED** [review]
🚀

**Review by Baroquem — APPROVED** [review]

**Comment by mhk33** [comment]
Sorry this is coming after it's been merged, but now that I see it in action, I like the icons but also find it a little busy. (Wet blanket, me.) I would suggest removing the ones in the Type of Change list, but ignore me if everyone prefers the icons to break up the text. :)
